app/services/CreditHistoryService.scala

```diff
@@ -70,11 +70,11 @@ class CreditHistoryService @Inject()(incomeTaxViewChangeConnector: IncomeTaxView
       creditModelForTaxYearPlusOne <- getCreditsByTaxYear(calendarYear + 1, nino)
     } yield (creditModelForTaxYear, creditModelForTaxYearPlusOne) match {
       case (Right(creditModelTY), Right(creditModelTYandOne)) =>
-        Right((creditModelTY ++ creditModelTYandOne).filter(_.date.getYear == calendarYear))
+        Right((creditModelTY ++ creditModelTYandOne).filter(_.documentDetail.taxYear == calendarYear.toString))
```

Comment:
what would be the reason to change type from Int to String? are we using YY-YY format or this is still just YYYY format ?